upup/pkg/fi/cloudup/awstasks/natgateway.go

```diff
@@ -209,19 +209,36 @@ func findNatGatewayFromRouteTable(cloud awsup.AWSCloud, routeTable *RouteTable)
 		}
 
 		if rt != nil {
+			clusterName, ok := routeTable.Tags[awsup.TagClusterName]
+			if !ok {
+				return nil, fmt.Errorf("Could not find '%s' tag from route table", awsup.TagClusterName)
+			}
 			var natGatewayIDs []*string
 			for _, route := range rt.Routes {
 				if route.NatGatewayId != nil {
 					natGatewayIDs = append(natGatewayIDs, route.NatGatewayId)
 				}
 			}
+			filteredNatGateways := []*ec2.NatGateway{}
+			for _, natGatewayID := range natGatewayIDs {
+				gw, err := findNatGatewayById(cloud, natGatewayID)
+				if err != nil {
+					return nil, err
+				}
 
-			if len(natGatewayIDs) == 0 {
-				klog.V(2).Infof("no NatGateway found in route table %s", *rt.RouteTableId)
-			} else if len(natGatewayIDs) > 1 {
-				return nil, fmt.Errorf("found multiple NatGateways in route table %s", *rt.RouteTableId)
+				for _, tag := range gw.Tags {
+					if fi.StringValue(tag.Key) == awsup.TagClusterName && fi.StringValue(tag.Value) == clusterName {
+						filteredNatGateways = append(filteredNatGateways, gw)
+						break
+					}
```

Comment:
I believe the new way to detect if a component belongs to the cluster is to check for `"kubernetes.io/cluster/" + clusterName`:
https://github.com/kubernetes/kops/blob/97869057129b30ea284c3ed1bdf1db36e752701d/pkg/resources/aws/tags.go#L27